core/server/data/xml/rss/index.js

```diff
@@ -104,18 +105,7 @@ function processUrls(html, siteUrl, itemUrl) {
             // if the relative URL begins with a '/' use the blog URL (including sub-directory)
             // as the base URL, otherwise use the post's URL.
             baseUrl = attributeValue[0] === '/' ? siteUrl : itemUrl;
-
```

Comment:
These next 6 lines of code are the main reason for this PR and the previous URL refactor. URLs can now be reliably created using `config.urlJoin`, which is tested.